documentation/src/main/asciidoc/userguide/chapters/query/hql/QueryLanguage.adoc

```diff
@@ -1604,63 +1725,87 @@ These functions are especially useful for reporting:
+`having` is applied after the groupings and aggregations have been done, while the `where` clause is applied before.
+
+[[hql-set-operators]]
+=== `union`, `intersect`, and `except`
+
+Query results may be combined using the operators:
+
+- `union` and `union all`,
+- `intersect` and `intersect all`, and
+- `except` and `except all`.
+
+Just like in SQL, `all` suppresses the elimination of duplicate results.
 
 [[hql-order-by]]
-=== The `ORDER BY` clause
+=== The `order by` clause
 
 By default, the results of the query are returned in an arbitrary order.
-The `ORDER BY` clause specifies a list of selected items used to order the results.
-The types of expressions considered valid as part of the `ORDER BY` clause include:
+The `order by` clause specifies a list of selected items used to order the results.
+The types of expressions considered valid as part of the `order by` clause include:
 
 * attributes of an entity or embeddable class,
 * scalar expressions such as arithmetic operations, functions, etc, and
 * identification variables declared in the select clause.
 
 [NOTE]
 ====
-The JPQL specification requires that all expressions occurring the `ORDER BY` clause must also occur in the `SELECT` clause.
+The JPQL specification requires that all expressions occurring the `order by` clause must also occur in the `select` clause.
 HQL does not enforce this restriction, but applications desiring database portability should be aware that some databases _do_.
 ====
 
 [[hql-order-by-example]]
-.Order by example
+//.Order by example
 ====
 [source, JAVA, indent=0]
 ----
 include::{sourcedir}/HQLTest.java[tags=hql-order-by-example]
 ----
 ====
 
-Each item listed in the `ORDER BY` clause may explicitly specify a direction, either:
+Each item listed in the `order by` clause may explicitly specify a direction, either:
 
-* `ASC` for ascending order, or
-* `DESC` for descending order.
+* `asc` for ascending order, or
+* `desc` for descending order.
 
 If no direction is explicitly specified, the results are returned in ascending order.
 
 Of course, there is an ambiguity with respect to null values.
 Therefore, the order of null values may also be explicitly specified:
 
-* `NULLS FIRST` puts null values at the beginning of the result set, and
-* `NULLS LAST` puts them last.
+* `nulls first` puts null values at the beginning of the result set, and
+* `nulls last` puts them last.
 
 In the next chapter we'll see a completely different way to write queries in Hibernate.
+
+[[hql-limit-offset]]
+=== The `limit` and `offset` clauses
+
+The `limit` and `offset` clauses are an alternative to the use of `setMaxResults()` and `setFirstResult()` respectively.
+
+[TIP]
+====
+If the `limit` or `offset` is parameterized, its much easier to use `setMaxResults()` or `setFirstResult()`.
+====
+
+The SQL syntax `fetch first ... rows only` and `fetch next ... rows only` is also allowed.
```

Comment:
It might be worth adding a paragraph about the `fetch` clause since it also allows to specify fetching a percentage of rows instead of just an absolute amount of rows and has the option _with ties_.

Reply:
I added a BNF last night.